linux-cachyos-hardened/PKGBUILD

```diff
@@ -143,17 +141,37 @@ _build_nvidia_open=${_build_nvidia_open-}
 # Build a debug package with non-stripped vmlinux
 _build_debug=${_build_debug-}
 
-if [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]] && [ "$_use_lto_suffix" = "y"  ]; then
-     _pkgsuffix="cachyos-${_cpusched}-lto"
-elif [ "$_use_llvm_lto" = "none" ]  && [ -z "$_use_kcfi" ] && [ "$_use_gcc_suffix" = "y" ]; then
-    _pkgsuffix="cachyos-${_cpusched}-gcc"
+# Enable AUTOFDO_CLANG for the first compilation to create a kernel, which can be used for profiling
+# Workflow:
+# https://cachyos.org/blog/2411-kernel-autofdo/
+# 1. Compile Kernel with _autofdo=y and _build_debug=y
+# 2. Boot the kernel in QEMU or on your system, see Workload
+# 3. Profile the kernel and convert the profile, see Generating the Profile for AutoFDO
+# 4. Put the profile into the sourcedir
+# 5. Run kernel build again with the _autofdo_profile_name path to profile specified
+_autofdo=${_autofdo-}
+
+# Name for the AutoFDO profile
+_autofdo_profile_name=${_autofdo_profile_name-}
+
+
+# ATTENTION: Do not modify after this line
+_is_lto_kernel() {
+    [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full" ]]
+    return $?
+}
+
+if _is_lto_kernel && [ "$_use_lto_suffix" = "y"  ]; then
+    _pkgsuffix=cachyos-hardened-lto
+elif ! _is_lto_kernel && [ "$_use_gcc_suffix" = "y" ]; then
+    _pkgsuffix=cachyos-hardened-gcc
```

Comment:
Why are you even hardcoding `hardened` here?

Reply:
Because it was not tagged with "-hardened" when I built it, so I added this, also, this is the PKGBUILD for the hardened kernel, why does it matter?

Reply:
It doesn't really matter much because -hardened uses its own configs, but ideally there should only be one PKGBUILD. We have much more in this repo to facilitate the building process to our repositories. Just to add another point, it keeps consistency :) with the rest of the PKGBUILDs that does this.